lib/gui/app/components/image-selector/image-selector.jsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const React = require('react')
+const ReactDOM = require('react-dom')
+const propTypes = require('prop-types')
+
+const middleEllipsis = require('./../../utils/middle-ellipsis')
+
+const { Provider } = require('rendition')
+
+const shared = require('/./../../../../../lib/shared/units')
+const { StepButton, StepNameButton, StepSelection,
+  Footer, Underline, DetailsText, ChangeButton } = require('./../../styled-components')
+
+const SelectImageButton = props => {
+
+  if (props.hasImage){
+    return (
+      <Provider>
+        <StepSelection>
+          <StepNameButton
+            plaintext
+            onClick={props.showSelectedImageDetails}
+            tooltip={props.imageBasename}
+          >
+            ( {middleEllipsis(props.imageName || props.imageBasename , 20)} )
+          </StepNameButton>
+          <DetailsText>
+            {shared.bytesToClosestUnit(props.imageSize)}
+          </DetailsText>
+          { props.flashing ?
+            null
+            :
+            <ChangeButton
+              plaintext
+              onClick={props.reselectImage}
+            >
+              Change
+            </ChangeButton>
+          }
+        </StepSelection>
+      </Provider>
+    )
+  }
+  else {
+    return (
+      <Provider>
+        <StepSelection>
+          <StepButton
+            primary
+            onClick={props.openImageSelector}
+          >
+            Select image
+          </StepButton>
+          <Footer>
+            { props.mainSupportedExtensions.join(', ') }, and
+            <Underline
+              tooltip={ props.extraSupportedExtensions.join(', ') }
+            >
+              {' '}others
```

Comment:
NIT: how about moving this space after the `and` (that's before the `<Underline>` element)?